src/parser/helper/Helper.ts

```diff
@@ -12,50 +13,87 @@ export function formatCaptures(tree, captures) {
+        if ((await fs.promises.lstat(sourcePath)).isFile()) {
+            const parseFile = checkAndGetFileExtension(sourcePath);
+            if (parseFile !== undefined && fileExtensionToGrammar.has(parseFile.fileExtension)) {
+                yield parseFile;
+            }
+        } else {
+            // sourcePath points to a directory, so use recursive function to find all files.
+
+            // Create set to improve lookup performance:
+            const excludedSet = new Set(excludedFolders);
+            // The folder at sourcePath itself cannot be excluded, so continue using delegating yield* generator call:
+            yield* findFilesAsyncRecursive(sourcePath, excludedSet);
+        }
+    } catch (e) {
+        console.error(e);
     }
+}
 
-    fs.readdirSync(dir).forEach((file) => {
-        const fileExtension = file.split(".").pop();
+async function* findFilesAsyncRecursive(
+    dir: string,
+    excludedFolders: Set<string>
+): AsyncGenerator<ParseFile> {
+    try {
+        const openedDir = await fs.promises.opendir(dir);
 
-        const currentPath = path.join(dir, file);
-        if (fs.statSync(currentPath).isDirectory()) {
-            const isPathExcluded = excludedFolders.some((folder) => {
-                return path.basename(currentPath) === folder;
-            });
-            if (isPathExcluded) {
-                return;
-            }
-            fileList = findFilesRecursively(
-                currentPath,
-                supportedFileExtensions,
-                excludedFolders,
-                fileList
-            );
-        } else if (fileExtension && supportedFileExtensions.includes(fileExtension.toLowerCase())) {
-            const parseFile = getParseFile(currentPath);
-            if (parseFile != undefined) {
-                fileList = fileList.concat(parseFile);
-            }
-        }
-    });
+        for await (const currentEntry of openedDir) {
+            const currentPath = path.join(dir, currentEntry.name);
 
-    return fileList;
+            if (currentEntry.isDirectory()) {
+                const isPathExcluded = excludedFolders.has(currentEntry.name);
+                if (!isPathExcluded) {
+                    // The current directory is not excluded, so recurse into subdirectory,
+                    // using delegating yield* generator call:
+                    yield* findFilesAsyncRecursive(currentPath, excludedFolders);
+                }
+            } // End of if (isDirectory)
+            else {
+                const parseFile = checkAndGetFileExtension(currentPath);
+                if (
+                    parseFile !== undefined &&
+                    fileExtensionToGrammar.has(parseFile.fileExtension)
+                ) {
+                    yield parseFile;
+                }
+            } // End of else (isDirectory)
+        } // End of for await (directory entries)
+    } catch (e) {
+        console.error(e);
+    }
```

Comment:
By using `try / catch` here and returning undefined, we likely would get multiple errors that are weird, in case an error happens in a deeply nested folder. I suggest we only ever handle this at the very top: so outside of this method.

Reply:
I have removed these try-catch-blocks inside the findFiles-functions and put a try-catch inside the (almost) top-level GenericParser.ts to catch any errors.